### PR TITLE
Fix e2e resource limits

### LINKED ISSUE #69989: Failing test: [sig-scheduling] SchedulerPriorities [Serial] Pod should be preferably scheduled to nodes which satisfy its limits
Testgrid: https://k8s-testgrid.appspot.com/sig-release-master-upgrade#gke-gci-new-gci-master-upgrade-cluster-new
https://k8s-testgrid.appspot.com/sig-release-master-blocking#gke-cos-master-serial

Example: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-gci-new-gci-master-upgrade-cluster-new/2362

This seems to be related to #69437

/sig scheduling
/kind failing-test
/priority important-soon

/cc @ravisantoshgudimetla

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What type of PR is this?**
/kind failing-test
/kind bug
/kind flake

**What this PR does / why we need it**:
While debugging

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Changes to fix 
#69989
- We need to add priorities explicitly to default algorithm provider. As of now, there is no way do it. So, added a function to do so.
- Updated e2e for limits priority function to be more robust.

I will monitor the test for some more time before closing the issue.

**Special notes for your reviewer**:
/cc @bsalamat 
/sig scheduling
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/factory/plugins.go [hunk 1]
```diff
@@ -167,6 +167,17 @@ func InsertPredicateKeyToAlgorithmProviderMap(key string) {
...
 
+// InsertPriorityKeyToAlgorithmProviderMap inserts a priority function to all algorithmProviders which are in algorithmProviderMap.
+func InsertPriorityKeyToAlgorithmProviderMap(key string) {
+	schedulerFactoryMutex.Lock()
+	defer schedulerFactoryMutex.Unlock()
+
+	for _, provider := range algorithmProviderMap {
+		provider.PriorityFunctionKeys.Insert(key)
+	}
+	return
```
Comment (Huang-Wei) [inline pkg/scheduler/factory/plugins.go]: Looks like no need to `return`

File: test/e2e/scheduling/priorities.go [hunk 2]
```diff
@@ -207,7 +209,8 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		}
 	})
 
-	It("Pod should perfer to scheduled to nodes pod can tolerate", func() {
+	It("Pod should preferable to scheduled to nodes pod can tolerate", func() {
```
Comment (Huang-Wei) [inline test/e2e/scheduling/priorities.go]: should -> should be, nodes -> nodes that

File: test/e2e/scheduling/priorities.go [hunk 3]
```diff
@@ -445,15 +453,16 @@ func addRandomTaitToNode(cs clientset.Interface, nodeName string) *v1.Taint {
 	return &testTaint
 }
 
-// updateMemoryOfNode updates the memory of given node with the given value
-func updateMemoryOfNode(c clientset.Interface, nodeName string, memory int64) error {
+// updateNodeAllocatable updates the allocatable values of given node with the given values.
+func updateNodeAllocatable(c clientset.Interface, nodeName string, memory int64, cpu int64) error {
```
Comment (Huang-Wei) [inline test/e2e/scheduling/priorities.go]: `memory int64, cpu int64` -> `memory, cpu int64`

Comment (Huang-Wei) [inline test/e2e/scheduling/priorities.go]:
```suggestion
func updateNodeAllocatable(c clientset.Interface, nodeName string, memory, cpu int64) error {
```

File: test/e2e/scheduling/priorities.go (hunk 2 above)
Comment (Huang-Wei) [inline test/e2e/scheduling/priorities.go]:
```suggestion
	It("Pod should be preferable to scheduled to nodes that pod can tolerate", func() {
```

File: pkg/scheduler/factory/plugins.go (hunk 1 above)
Comment (ravisantoshgudimetla) [reply pkg/scheduler/factory/plugins.go]: Just a coding style to indicate end of function, I don't have a preference though.

File: test/e2e/scheduling/priorities.go (hunk 2 above)
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: Yeah, I will fix it. Thanks.

File: test/e2e/scheduling/priorities.go (hunk 3 above)
Comment (ravisantoshgudimetla) [reply test/e2e/scheduling/priorities.go]: Style issue again. I don't mind changing it but AFAIK there is no standard preference I think.

**Review by ravisantoshgudimetla — COMMENTED** [review]
Thanks for review @Huang-Wei. I will address them.

File: pkg/scheduler/algorithmprovider/defaults/defaults.go [hunk 4]
```diff
@@ -212,9 +212,11 @@ func ApplyFeatureGates() {
 
 	// Prioritizes nodes that satisfy pod's resource limits
 	if utilfeature.DefaultFeatureGate.Enabled(features.ResourceLimitsPriorityFunction) {
+		glog.Warningf("Registering resourcelimits priority function")
```
Comment (bsalamat) [inline pkg/scheduler/algorithmprovider/defaults/defaults.go]: Why is this a warning instead of an info log?

**Review by bsalamat — COMMENTED** [review]
Looks good overall. Just a minor comment.

File: pkg/scheduler/algorithmprovider/defaults/defaults.go (hunk 4 above)
Comment (ravisantoshgudimetla) [reply pkg/scheduler/algorithmprovider/defaults/defaults.go]: It was copy-paste error as I copied this from earlier block. I will change it. Thanks.

Comment (ravisantoshgudimetla) [reply pkg/scheduler/algorithmprovider/defaults/defaults.go]: Done.

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**